gensim/models/ensemblelda.py

```diff
@@ -0,0 +1,1268 @@
+            Default: None, which uses 0.11 for
+            masking_method "rank", and 0.95
+            for "mass".
+        distance_workers : number, optional
+            When distance_workers is None, it defaults to
+            os.cpu_count() for maximum performance.
+            Default is 1, which is not multiprocessed.
+            Set to > 1 to enable multiprocessing.
+        **gensim_kw_args
+            All the parameters for the gensim model,
+            that is used for each model in the ensemble,
+            can be provided as additional keyword arguments.
+            https://radimrehurek.com/gensim/models/ldamodel.html
+
+        """
+
+        if "id2word" not in gensim_kw_args:
+            gensim_kw_args["id2word"] = None
+        if "corpus" not in gensim_kw_args:
+            gensim_kw_args["corpus"] = None
+
+        # dictionary. modified version of from gensim/models/ldamodel.py
+        # error messages are copied from the original gensim module [1]:
+        # will create a fake dict if no dict is provided.
+        if gensim_kw_args["id2word"] is None and not gensim_kw_args["corpus"] is None:
+            logger.warning("no word id mapping provided; initializing from corpus, assuming identity")
+            gensim_kw_args["id2word"] = utils.dict_from_corpus(gensim_kw_args["corpus"])
+        if gensim_kw_args["id2word"] is None and gensim_kw_args["corpus"] is None:
+            raise ValueError("at least one of corpus/id2word must be specified, to establish "
+                             "input space dimensionality. Corpus should be provided using the "
+                             "keyword argument corpus.")
+
+        if type(topic_model_kind) == str:
+            self.topic_model_kind = {
+                "lda": ldamodel.LdaModel,
+                "ldamulticore": ldamulticore.LdaMulticore
+            }[topic_model_kind]
+        else:
+            self.topic_model_kind = topic_model_kind
+
+        # Store some of the parameters
+        self.num_models = num_models
+        self.gensim_kw_args = gensim_kw_args
+
+        # some settings affecting performance
+        self.memory_friendly_ttda = memory_friendly_ttda
+        self.distance_workers = distance_workers
+
+        # this will provide the gensim api to the ensemble basically
+        self.classic_model_representation = None
+
+        # the ensembles state
+        self.random_state = utils.get_random_state(random_state)
+        self.sstats_sum = 0
+        self.eta = None
+        self.tms = []
+        # initialize empty topic term distribution array
+        self.ttda = np.empty((0, len(gensim_kw_args["id2word"])))
+        self.asymmetric_distance_matrix_outdated = True
+
+        # in case the model will not train due to some
+        # parameters, stop here and don't train.
+        if num_models <= 0:
+            return
+        if ("corpus" not in gensim_kw_args):
+            return
+        if gensim_kw_args["corpus"] is None:
```

Comment:
You can combine this with the above conditional without losing any readibility.

```suggestion
        if gensim_kw_args.get("corpus") is None:
```

Reply:
done